[Bug]: test_snapshot_restore_with_different_access_mode failed

### Description
Failed test case is `tests/functional/pv/pvc_snapshot/test_snapshot_restore_with_different_access_mode.py::TestSnapshotRestoreWithDifferentAccessMode::test_snapshot_restore_with_different_access_mode`


```


          pod.verify_data_integrity(
                pod_obj,
                file_name_pod,
                pod_obj.pvc.md5sum,
                pod_obj.pvc.data["spec"]["volumeMode"]
                == constants.VOLUME_MODE_BLOCK,
            )

tests/functional/pv/pvc_snapshot/test_snapshot_restore_with_different_access_mode.py:220: 



pod_obj = <ocs_ci.ocs.resources.pod.Pod object at 0x7fe9244def90>
file_name = '/dev/rbdblock'
original_md5sum = '5bd09783c18f31811030890c4bd2afd9', block = True


def verify_data_integrity(pod_obj, file_name, original_md5sum, block=False):
    """
    Verifies existence and md5sum of file created from first pod

    Args:
        pod_obj (Pod): The object of the pod
        file_name (str): The name of the file for which md5sum to be calculated
        original_md5sum (str): The original md5sum of the file
        block (bool): True if the volume mode of PVC used on pod is 'Block'.
            file_name will be the devicePath in this case.

    Returns:
        bool: True if the file exists and md5sum matches

    Raises:
        AssertionError: If file doesn't exist or md5sum mismatch
    """
    file_path = file_name if block else get_file_path(pod_obj, file_name)
    assert check_file_existence(pod_obj, file_path), f"File {file_name} doesn't exists"
    current_md5sum = cal_md5sum(pod_obj, file_name, block)
    logger.info(f"Original md5sum of file: {original_md5sum}")
    logger.info(f"Current md5sum of file: {current_md5sum}")

  assert current_md5sum == original_md5sum, "Data corruption found"
E       AssertionError: Data corruption found



ocs_ci/ocs/resources/pod.py:1436: AssertionError
```


### Steps to Reproduce
Run the test case tests/functional/pv/pvc_snapshot/test_snapshot_restore_with_different_access_mode.py::TestSnapshotRestoreWithDifferentAccessMode::test_snapshot_restore_with_different_access_mode

Failure is rare.

### Actual Behavior
Test case failed due to md5sum mismatch

### Expected Behavior
md5sum matches

### Impact (likelihood of reproduction, impact on the cluster and on other tests, etc.)
Issue found twice in 100+ runs.
No impact on the cluster or other tests.
### Screenshots (if applicable)

### Environment
- Test Suite(s): tier1
- Platform(s): VSphere (not dependent on platform) 
- Version(s): ODF, OCP 4.20
- OS:

### Additional Context

--- a/tests/functional/pv/pvc_snapshot/test_snapshot_restore_with_different_access_mode.py
+++ b/tests/functional/pv/pvc_snapshot/test_snapshot_restore_with_different_access_mode.py
@@ -70,20 +70,20 @@ class TestSnapshotRestoreWithDifferentAccessMode(ManageTest):
 
         # Start IO
         log.info("Starting IO on all pods")
+
         for pod_obj in self.pods:
-            if pod_obj.pvc.volume_mode == constants.VOLUME_MODE_BLOCK:
-                storage_type = "block"
-                io_direct = 1
-            else:
-                storage_type = "filesystem"
-                io_direct = 0
+            storage_type = (
+                "block"
+                if pod_obj.pvc.volume_mode == constants.VOLUME_MODE_BLOCK
+                else "fs"
+            )
             pod_obj.run_io(
                 storage_type=storage_type,
                 size="1G",
                 runtime=20,
                 fio_filename=file_name,
                 end_fsync=1,
-                direct=io_direct,
+                direct=int(pod_obj.pvc.volume_mode == constants.VOLUME_MODE_BLOCK),
             )
             log.info(f"IO started on pod {pod_obj.name}")
         log.info("Started IO on all pods")

